dom.bs

```diff
@@ -1413,6 +1413,152 @@ can only be used to influence an ongoing one.
 
 
 
+<h2 id=aborting-ongoing-activities>Aborting ongoing activities</h3>
+
+<p>Though promises do not have a built-in aborting mechanism, many APIs using them require abort
+semantics. {{AbortController}} is meant to support these requirements by providing an
+{{AbortController/abort()}} method that toggles the state of a corresponding {{AbortSignal}} object.
+The API which wishes to support aborting can accept an {{AbortSignal}} object, and use its state to
+determine how to proceed.
+
+<p>APIs that rely upon {{AbortController}} are encouraged to respond to {{AbortController/abort()}}
+by rejecting any unsettled promise with a new {{DOMException}} with [=error name=] "{{AbortError}}".
+
+<div class=example id=aborting-ongoing-activities-example>
+ <p>A hypothetical <code>doAmazingness({ ... })</code> method could accept an {{AbortSignal}} object
+ in order to support aborting as follows:
+
+ <pre><code class=lang-javascript>
+const controller = new AbortController();
+const signal = controller.signal;
+
+startSpinner();
+
+doAmazingness({ ..., signal })
+  .then(result => ...)
+  .catch(err => {
+    if (err.name == 'AbortError') return;
+    showUserErrorMessage();
+  })
+  .then(() => stopSpinner());
+
+// &hellip;
+
+controller.abort();</code></pre>
+
+ <p><code>doAmazingness</code> could be implemented as follows:
+
+ <pre><code class=lang-javascript>
+function doAmazingness({signal}) {
+  return new Promise((resolve, reject) => {
+    // Begin doing amazingness, and call resolve(result) when done.
+    // But also, watch for signals:
+    signal.addEventListener('abort', () => {
+      // Stop doing amazingness, and:
+      reject(new DOMException('Aborted', 'AbortError'));
+    });
+  });
+}
+</code></pre>
+
+ <p>APIs that require more granular control could extend both {{AbortController}} and
+ {{AbortSignal}} objects according to their needs.
+</div>
+
+
+<h3 id=interface-abortcontroller>Interface {{AbortController}}</h3>
+
+<pre class="idl">
+[Constructor,
+ Exposed=(Window,Worker)]
+interface AbortController {
+  [SameObject] readonly attribute AbortSignal signal;
+
+  void abort();
+};</pre>
+
+<dl class=domintro>
+ <dt><code><var>controller</var> = new <a constructor lt="AbortController()">AbortController</a>()</code>
+ <dd>Returns a new <var>controller</var> whose {{AbortController/signal}} is set to a newly
+ created {{AbortSignal}} object.
+
+ <dt><code><var>controller</var> . </code>{{AbortController/signal}}
+ <dd>Returns the {{AbortSignal}} object associated with this object.
+
+ <dt><code><var>controller</var> . </code>{{AbortController/abort()}}
+ <dd>Invoking this method will set this object's {{AbortSignal}}'s [=AbortSignal/aborted flag=],
+ thereby signaling to any observers that the associated activity should be aborted.
```

Comment:
Nit: I think this would be clearer as "... set this AbortSingal's aborted flag, and signal to any observers..." (instead of using "thereby")